ncoreparser/client.py

```diff
@@ -65,11 +64,11 @@ def login(self, username, password):
     @_check_login
     # pylint: disable=too-many-arguments
     def search(self, pattern, type=SearchParamType.ALL_OWN, where=SearchParamWhere.NAME,
-               sort_by=ParamSort.UPLOAD, sort_order=ParamSeq.DECREASING, number=TORRENTS_PER_PAGE):
-        page_count = 0
+               sort_by=ParamSort.UPLOAD, sort_order=ParamSeq.DECREASING, number=None):
+        page_count = 1
         torrents = []
-        while page_count * TORRENTS_PER_PAGE < number:
-            url = URLs.DOWNLOAD_PATTERN.value.format(page=page_count + 1,
+        while number is None or len(torrents) < number:
```

Comment:
This thing may occur in an infinite loop.
If we keep the `number` variable `None` (because it is the default)
`number is None` will be true, and unchanged.

Reply:
I put a `number is None` check in the if at the end of the loop on line number 82.

Reply:
Ohh okay, you are right. I missed it. 😄 
I will run a test on it today. If it works I can merge it.
Also please increase the version in `pyproject.toml` to `2.1.1`